README.md
# flutter_vslider

[![Open Source Love](https://badges.frapsoft.com/os/v3/open-source.svg?v=103)](https://github.com/ellerbrock/open-source-badges/)

[![forthebadge](https://forthebadge.com/images/badges/built-for-android.svg)](https://forthebadge.com)


## Usage
to change a value


![gif](misc/Demo_20190927163224.gif)


## Example

```dart
void main() => runApp(MyApp());

class MyApp extends StatefulWidget {
  @override
  _MyAppState createState() => _MyAppState();
}

class _MyAppState extends State<MyApp> with SingleTickerProviderStateMixin{


  double value = 0.5;
  double elevation = 3.0;
  double iconOpacity = 0.25;
  double iconSize = 0.0;
  double shadowRadius = 100.0;

//animation stuff
  AnimationController controller;
  Animation animation;

  @override
  void initState() {
    super.initState();

    controller = AnimationController(
      duration: Duration(milliseconds: 100),
      vsync: this,
    );

    //https://api.flutter.dev/flutter/animation/Curves-class.html
    animation = CurvedAnimation(parent: controller,curve: Curves.easeIn);


    animation.addStatusListener((status){
      print(status);
    });

    controller.addListener((){
      setState(() {
        iconSize = animation.value;
      });
//      print(controller.value);
      print('animation ${animation.value}');
    });
  }

  @override
  void dispose() {
    controller.dispose();
    super.dispose();
  }
  //end animation stuff


  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        appBar: AppBar(
          title: Text('vSlider'),
          backgroundColor: Colors.blueGrey,
        ),
        body: Center(
          child: vSlider(
            color: Colors.blueGrey[500],
            fillColor: Colors.lightGreenAccent,
            fillShadowColor: Colors.lightGreenAccent[700],
            fillShadowRadius: shadowRadius,
            elevation: elevation,
            radius: 25.0,
            value: value,
            onSlide: (v){
              print('onSlide ${-v.delta.dy}');
              value += -v.delta.dy/300.0;
              setState(() {
                value = value.clamp(0.0, 1.0);
              });
            },
            onStart: (details){
              print('onStart $details');
              controller.forward(from: 0);
              setState(() {
                elevation = 24.0;
              });
            },
            onEnd: (details){
              print('onEnd $details');
              controller.reverse(from: 1);
              setState(() {
                elevation = 3.0;
              });
            },
            icon: Icon(
              Icons.music_note,
              size: (iconSize * 10.0) + 50.0,
              color: Colors.black.withOpacity(0.25),
            ),
          ),
        ),
      ),
    );
  }
}

```


## [buy me a ☕](paypal.me/JGarza9788)


CHANGELOG.md
## 0.1.3

* use AnimatedContainer

## 0.1.2

* fixed UpperCamelCase

## 0.1.1

* minor tweaks

## 0.1.0

* Initial Open Source release.

test/widget_test.dart
import 'package:flutter/material.dart';
import 'package:flutter_vslider/flutter_vslider.dart';

void main() => runApp(MyApp());

class MyApp extends StatefulWidget {
  @override
  _MyAppState createState() => _MyAppState();
}

class _MyAppState extends State<MyApp> with SingleTickerProviderStateMixin{


  double value = 0.5;
  double elevation = 3.0;
  double iconOpacity = 0.25;
  double iconSize = 0.0;
  double shadowRadius = 100.0;





  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        appBar: AppBar(
          title: Text('vSlider'),
          backgroundColor: Colors.blueGrey,
        ),
        body: Center(
          child: VSlider(
            color: Colors.blueGrey[500],
            fillColor: Colors.lightGreenAccent,
            fillShadowColor: Colors.lightGreenAccent[700],
            fillShadowRadius: shadowRadius,
            elevation: elevation,
            radius: 25.0,
            value: value,
            onSlide: (v){
              print('onSlide ${-v.delta.dy}');
              value += -v.delta.dy/300.0;
              setState(() {
                value = value.clamp(0.0, 1.0);
              });
            },
            onStart: (details){
              print('onStart $details');
              setState(() {
                elevation = 24.0;
              });
            },
            onEnd: (details){
              print('onEnd $details');
              setState(() {
                elevation = 3.0;
              });
            },
            icon: Icon(
              Icons.music_note,
              size: (iconSize * 10.0) + 50.0,
              color: Colors.black.withOpacity(0.25),
            ),
          ),
        ),
      ),
    );
  }
}




test/flutter_vslider_test.dart
import 'package:flutter_test/flutter_test.dart';

import 'package:flutter_vslider/flutter_vslider.dart';

void main() {
  test('adds one to input values', () {
    final calculator = Calculator();
    expect(calculator.addOne(2), 3);
    expect(calculator.addOne(-7), -6);
    expect(calculator.addOne(0), 1);
    expect(() => calculator.addOne(null), throwsNoSuchMethodError);
  });
}


lib/flutter_vslider.dart
import 'package:flutter/material.dart';

class VSlider extends StatelessWidget {
  VSlider({
    this.color,
    this.fillColor,
    this.fillShadowRadius,
    this.fillShadowColor,
    this.elevation,
    this.radius,
    this.value,
    this.icon,
    @required this.onSlide,
    this.onStart,
    this.onEnd,
  });

  final Color color;
  final Color fillColor;
  final Color fillShadowColor;
  final double fillShadowRadius;
  final double value;
  final double elevation;
  final double radius;
  final Widget icon;
  final Function onSlide;
  final Function onStart;
  final Function onEnd;

  @override
  Widget build(BuildContext context) {
    return GestureDetector(
//      onVerticalDragUpdate: (v){
////        print(v);
//        print(v.delta.dy/300.0);
////        this.value = v.localPosition.dy/300.0;
//      },
      onVerticalDragUpdate: onSlide,
      onVerticalDragStart: onStart ,
      onVerticalDragEnd: onEnd ,

      child: AnimatedContainer(
        duration: Duration(milliseconds: 750),
        curve: Curves.easeInOut,
        child: Material(

          clipBehavior: Clip.hardEdge,
          elevation: elevation==null? 1.0:elevation,
          borderRadius: BorderRadius.circular(radius==null?25.0:radius),

          child: Container(
            height: 300.0,
            width: 100.0,
            child: Stack(
              overflow: Overflow.clip,
              alignment: Alignment.bottomCenter,
              children: <Widget>[
                Container(
                  color: color,
                ),

                Container(
//                color: fillColor,
                  height: this.value * 300.0,
                  decoration: BoxDecoration(
                    color: fillColor,
                    boxShadow: [
                      new BoxShadow(
                        color: fillShadowColor==null?Colors.black:fillShadowColor,
                        offset: Offset(0.0, 0.0),
                        blurRadius: fillShadowRadius==null?0.0:fillShadowRadius,
                      )
                    ],
                  ),
                ),

                Align(
                  alignment: Alignment.bottomCenter,
                  child: Column(
                    mainAxisAlignment: MainAxisAlignment.end,
                    children: <Widget>[
                      Container(
                        height: 100.0,
                        width: 100.0,
                        child: icon==null?Container():icon,
                      )
                    ],
                  ),
                )
              ],
            ),
          ),
        ),
      ),

    );
  }
}


example/main.dart
import 'package:flutter/material.dart';
import '../lib/flutter_vslider.dart';

void main() => runApp(MyApp());

class MyApp extends StatefulWidget {
  @override
  _MyAppState createState() => _MyAppState();
}

class _MyAppState extends State<MyApp> with SingleTickerProviderStateMixin{


  double value = 0.5;
  double elevation = 3.0;
  double iconOpacity = 0.25;
  double iconSize = 0.0;
  double shadowRadius = 100.0;


  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        appBar: AppBar(
          title: Text('vSlider'),
          backgroundColor: Colors.blueGrey,
        ),
        body: Center(
          child: VSlider(
            color: Colors.blueGrey[500],
            fillColor: Colors.lightGreenAccent,
            fillShadowColor: Colors.lightGreenAccent[700],
            fillShadowRadius: shadowRadius,
            elevation: elevation,
            radius: 25.0,
            value: value,
            onSlide: (v){
              print('onSlide ${-v.delta.dy}');
              value += -v.delta.dy/300.0;
              setState(() {
                value = value.clamp(0.0, 1.0);
              });
            },
            onStart: (details){
              print('onStart $details');
              setState(() {
                elevation = 24.0;

//                shadowRadius = 100.0;
              });
            },
            onEnd: (details){
              print('onEnd $details');
              setState(() {
                elevation = 3.0;

//                shadowRadius = 0.0;
              });
            },
            icon: Icon(
              Icons.music_note,
              size: (value * 10.0) + 50.0,
              color: Colors.black.withOpacity((value * 0.50) + 0.25),
            ),
          ),
        ),
      ),
    );
  }
}


